<a href="https://colab.research.google.com/github/100485446-UC3M/Primera_Practica_Machine_Learning/blob/main/Predicciones.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Predicciones Práctica 1
## Esteban Gómez Buitrago 100485446
## Nicolás Alejandro Cuesta García 100495966

## Análisis Exploratorio de Datos

PAra realizar este projecto utilizamos los set de datos "attrition_availabledata_10.csv" y"attrition_competition_10.csv". Decidimos utilizar estos, después de realizar la suma de los últimos dos digitos del NIA 100485446 (4 + 6 = 10).

Al analizar los datos, la primera conclusión a la que se puede llegar es que este es un problema de clasificación. Se está realizando un estudio en el cual a través de algunas variables se va a definir si la clase (en este caso, el attributo "Attrition") es "Si" o "No".

Para encontrar ciertos datos importantes para continuar con el análisis, será necesario utilizar ciertas funciones de python. El código y los resultados están descritos a continuación:

In [ ]:
#Las librerías necesarias son importadas
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [18]:
#Se crea una semilla para poder replicar los resultado y se leen los datos
np.random.seed(100495966)
data = pd.read_csv('attrition_availabledata_10.csv')

#Para chequear que sea el set de datos correcto
#print(data.head())

print(f"Número de variables (columnas): {data.shape[1] - 1}")
print(f"Número de instancias (filas): {data.shape[0]}")
print("La cantidad de datos vacíos por columna es: ")
data.isnull().sum()




Número de variables (columnas): 30
Número de instancias (filas): 2940
La cantidad de datos vacíos por columna es: 


hrs                         0
absences                    0
JobInvolvement              0
PerformanceRating           0
EnvironmentSatisfaction    15
JobSatisfaction            12
WorkLifeBalance            29
Age                         0
BusinessTravel              0
Department                  0
DistanceFromHome            0
Education                   0
EducationField              0
EmployeeCount               0
EmployeeID                  0
Gender                      0
JobLevel                    0
JobRole                     0
MaritalStatus               0
MonthlyIncome               0
NumCompaniesWorked         17
Over18                      0
PercentSalaryHike           0
StandardHours               0
StockOptionLevel            0
TotalWorkingYears           5
TrainingTimesLastYear       0
YearsAtCompany              0
YearsSinceLastPromotion     0
YearsWithCurrManager        0
Attrition                   0
dtype: int64

Se puede afirmar que el estudio será realizado con 30 variables y 2940 instancias distintas. Es importante esaltar que asl número de variables se le resto la variable "Attrition" ya que esta es nuestra variable de calisifcación (objetivo). Después de analizar las variable individualmente encontramos que tenemos variables catégoricas, ordinales y númericas.

Las variabes catégoricas son (7):   

*   BusinessTravel
*   Department
*   EducationField
*   Gender
*   JobRole
*   MaritalStatus
*   Over18

Las númericas incluyen (15):  

*   hrs
*   absences
*   Age
*   DistanceFromHome
*   MonthlyIncome
*   PercentSalaryHike
*   TotalWorkingYears
*   TrainingTimesLastYear
*   NumCompaniesWorked
*   EmployeeCount
*   EmployeeID
*   StandardHours
*   YearsAtCompany
*   YearsSinceLastPromotion
*   YearsWithCurrManager

Finalmente, entre las ordinales están (8):

*   JobInvolvement
*   JobLevel
*   Education
*   PerformanceRating
*   EnvironmentalSatisfaction
*   JobSatisfaction
*   WorkLifeBalance
*   StockOptionLevel


Entre todas estas instancias tenemos valores vacíos en 5 variables.